### PR TITLE
[No Jira] [Rust] fix links to spec

### PR DESCRIPTION
**Pull request by xxchan**
## What is the purpose of the change

fix doc link

Previous links cannot jump to the sections any more.

e.g.,
old: https://avro.apache.org/docs/current/spec.html#single_object_encoding
new: https://avro.apache.org/docs/current/specification/#single-object-encoding

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Documentation

- Does this pull request introduce a new feature? no

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]
Looks good to me.

**Comment by Xuanwo** [comment]
Hi @fokko, This is just a minor document update. Could you take a look and review it? Thanks!

**Comment by martin-g** [comment]
I will handle the PR once the 1.12.0/0.17.0 release is done!
The tag is already cut, so merging it now won't change anything for the rustdocs of 0.17.0...

**Comment by Xuanwo** [comment]
> I will handle the PR once the 1.12.0/0.17.0 release is done! The tag is already cut, so merging it now won't change anything for the rustdocs of 0.17.0...

Hi @martin-g, thanks a lot for maintaining this. I'm a bit surprised to see the PR left open as it is. It could become outdated, conflicted, and inactive. Are there any other concerns preventing us from merging?

**Comment by xxchan** [comment]
Hi @martin-g, thanks for the reply, but I'm a little confused: Do you mean we want or not want this into 0.17.0? I think actually either option wouldn't prevent us from merging this trivial PR. 🤣

Or do you just mean you are busy with releasing, and want to handle PRs only after the release is done?

**Comment by martin-g** [comment]
We can't have it in 0.17.0 unless the vote is cancelled for some serious reason. The broken links in rustdoc is not a reason to cancel the vote!
Merging it now or after few days does not make much difference. The only question is to which other branches to backport it too. This is not very clear at the moment!

> Or do you just mean you are busy with releasing, and want to handle PRs only after the release is done?

I am busy with my daily job and my personal life! 
Avro is a hobby for me! I work on it when I want to do it!


> It could become outdated, conflicted, and inactive

The past does not show any evidence of such kind of problems (in the Rust SDK)!

File: lang/rust/avro/src/schema.rs [hunk 1]
```diff
@@ -1022,7 +1022,7 @@ impl Schema {
     /// Converts `self` into its [Parsing Canonical Form].
     ///
     /// [Parsing Canonical Form]:
-    /// https://avro.apache.org/docs/1.8.2/spec.html#Parsing+Canonical+Form+for+Schemas
+    /// https://avro.apache.org/docs/1.8.2/specification/#parsing-canonical-form-for-schemas
```
Comment (Fokko) [inline lang/rust/avro/src/schema.rs:1025]:
```suggestion
    /// https://avro.apache.org/docs/current/specification/#parsing-canonical-form-for-schemas
```

File: lang/rust/avro/src/schema.rs [hunk 2]
```diff
@@ -1032,9 +1032,9 @@ impl Schema {
     /// Generate [fingerprint] of Schema's [Parsing Canonical Form].
     ///
     /// [Parsing Canonical Form]:
-    /// https://avro.apache.org/docs/1.8.2/spec.html#Parsing+Canonical+Form+for+Schemas
+    /// https://avro.apache.org/docs/1.8.2/specification/#parsing-canonical-form-for-schemas
```
Comment (Fokko) [inline lang/rust/avro/src/schema.rs:1035]:
```suggestion
    /// https://avro.apache.org/docs/current/specification/#parsing-canonical-form-for-schemas
```

File: lang/rust/avro/src/schema.rs [hunk 3]
```diff
@@ -2143,7 +2143,7 @@ impl Serialize for RecordField {
 }
 
 /// Parses a **valid** avro schema into the Parsing Canonical Form.
-/// https://avro.apache.org/docs/1.8.2/spec.html#Parsing+Canonical+Form+for+Schemas
+/// https://avro.apache.org/docs/1.8.2/specification/#parsing-canonical-form-for-schemas
```
Comment (Fokko) [inline lang/rust/avro/src/schema.rs:2146]:
```suggestion
/// https://avro.apache.org/docs/current/specification/#parsing-canonical-form-for-schemas
```

**Review by Fokko — APPROVED** [review]
Found three broken links, but apart it looks good. Thanks for fixing this 👍  I think we can move this forward. If we need another release candidate, then we can cherry-pick this into 0.17.0.

**Comment by xxchan** [comment]
Thanks again for you all. ❤️ 

Personally I don't mind too much whether this can be cherry-picked into 0.17. (It's definitely good to also have this if we have other cherry-picks though.)

And no mean to put pressure, but just think "merge earlier" can save our energy 😄

**Review by martin-g — APPROVED** [review]

**Pull request merged**